packages/server-wallet/src/config/types.ts

```diff
@@ -0,0 +1,92 @@
+import {Level} from 'pino';
+
+/**
+ * Either a connection string or a config object with the dbName specified
+ */
+export type DatabaseConnectionConfiguration = RequiredConnectionConfiguration &
+  Partial<OptionalConnectionConfiguration>;
+
+/**
+ * Either a database connection string or a config object specifying the database name
+ */
+export type RequiredConnectionConfiguration = {dbName: string} | string;
+
+/**
+ * Optional database config properties.
+ */
+export type OptionalConnectionConfiguration =
+  | {host: string; port: number; user?: string; password?: string}
+  | string;
+
+/**
+ * Database pool size options
+ */
+export type DatabasePoolConfiguration = {max?: number; min?: number};
+
+/**
+ * The minimum required database configuration that must be provided
+ */
+export type RequiredDatabaseConfiguration = {connection: RequiredConnectionConfiguration};
+export type OptionalDatabaseConfiguration = {
+  pool?: DatabasePoolConfiguration;
+  debug?: boolean;
+  connection: OptionalConnectionConfiguration;
+};
+
+/**
+ * The fully defined database configuration
+ */
+export type DatabaseConfiguration = RequiredDatabaseConfiguration & OptionalDatabaseConfiguration;
+
+/**
+ * Logging configuration options
+ */
+export type LoggingConfiguration = {logLevel: Level; logDestination: string};
+
+/**
+ * Metrics configuration options
+ */
+export type MetricConfiguration = {timingMetrics: boolean; metricsOutputFile?: string};
+
+/**
+ * The minimum required configuration to use the server wallet.
+ */
+export type RequiredServerWalletConfig = {
+  databaseConfiguration: RequiredDatabaseConfiguration;
+};
+
+/**
+ * Additional configuration options for the server wallet that are not required.
+ */
+export interface OptionalServerWalletConfig {
```

Comment:
I think some of these config variables should be required, such as `networkConfiguration`.